source/fundamentals/crud/read-operations/change-streams.txt

```diff
@@ -0,0 +1,306 @@
+     - | The collation to use for the change stream cursor.
+
+   * - ``Comment``
+     - | A comment to attach to the operation.
+
+.. _csharp-change-stream-pre-post-image:
+
+Include Pre-Images and Post-Images
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. important::
+
+   You can enable pre-images and post-images on collections only if your
+   deployment uses MongoDB v6.0 or later.
+
+By default, when you perform an operation on a collection, the
+corresponding change event includes only the delta of the fields
+modified by that operation. To see the full document before or after a
+change, create a ``ChangeStreamOptions`` object and specify the
+``FullDocumentBeforeChange`` or the ``FullDocument`` options. Then, pass the
+``ChangeStreamOptions`` object to the ``Watch()`` or ``WatchAsync()`` method.
+
+The **pre-image** is the full version of a document *before* a change. To include the
+pre-image in the change stream event, set the ``FullDocumentBeforeChange``
+option to one of the following values:
+
+- ``ChangeStreamFullDocumentBeforeChangeOption.WhenAvailable``: The change event
+  includes a pre-image of the modified document for change events only if the
+  pre-image is available.
+- ``ChangeStreamFullDocumentBeforeChangeOption.Required``:
+  The change event includes a pre-image of the modified document for change
+  events. If the pre-image is not available, the driver raises an error.
+
+The **post-image** is the full version of a document *after* a change. To include the
+post-image in the change stream event, set the ``FullDocument`` option to
+one of the following values:
+
+- ``ChangeStreamFullDocumentOption.UpdateLookup``: The change event includes a
+  copy of the entire changed document from some time after the change.
+- ``ChangeStreamFullDocumentOption.WhenAvailable``: The change event includes a
+  post-image of the modified document for change events only if the post-image
+  is available.
+- ``ChangeStreamFullDocumentOption.Required``: The change event includes a
+  post-image of the modified document for change events. If the post-image is
+  not available, the driver raises an error.
+
+The following example opens a change stream on a collection and includes the post-image
+of updated documents by specifying the ``FullDocument`` option. Select the
+:guilabel:`Asynchronous` or :guilabel:`Synchronous` tab to see the corresponding
+code.
+
+.. tabs::
+
+   .. tab:: Asynchronous
+      :tabid: change-stream-async
+
+      .. literalinclude:: /includes/code-examples/change-streams/change-streams.cs
+         :start-after: start-change-stream-post-image-async
+         :end-before: end-change-stream-post-image-async
+         :language: csharp
+
+   .. tab:: Synchronous
+      :tabid: change-stream-sync
+
+      .. literalinclude:: /includes/code-examples/change-streams/change-streams.cs
+         :start-after: start-change-stream-post-image
+         :end-before: end-change-stream-post-image
+         :language: csharp
+
```

Comment:
S: I think it would be helpful to see what the post-image looks like in the output

Reply:
Added!